.github/workflows/test.yml

```diff
@@ -44,6 +54,7 @@ jobs:
         run: pytest --cov
 
       - name: Coverage
+        if: ${{ success() &&  matrix.python == env.PYTHON_COVERAGE_ENABLED }}
```

Comment:
`success()` is optional, but we probably don't care if the build failed. If it goes ahead, we could have a single point in the coverage graph where coverage was reported as `0` or lower than the expected.

Also, unless we limit when we upload the coverage, the build in Codecov will have multiple XML's uploaded, which is not necessary, and may cause troubles if the coverage module is not working as expected on win/mac/linux. I think for CI we can pick linux + python stable or the python version most of us are using? :man_shrugging: